src/core/registry/registry.test.ts

```diff
@@ -647,6 +648,40 @@ describe('registry', () => {
         headers: [`Authorization: SUPERFACE-SDK-TOKEN ${TEST_SDK_TOKEN}`],
       });
     });
+
+    it('throws error on invalid status code and response body with detail', async () => {
+      const mockBody = {
+        detail: 'Test',
+        title: 'Title',
+      };
+      const mockResponse = {
+        statusCode: 400,
+        body: mockBody,
+        headers: { test: 'test' },
```

Comment:
question: I don't know if relying only on body structure is right here. Server is returning header`Content-Type: application/problem+json` so the clients know what structure to expect.

Reply:
I can use `content-type` to decide.